### PR TITLE
Add migrator for CUDA on Windows

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

PR created in the context of bringing `nvcc` on Windows. Related PRs: 
* https://github.com/conda-forge/nvcc-feedstock/pull/49
* https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/129

### CODE REVIEW

File: recipe/migrations/window_cuda.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  migration_number:
+    1
+  build_number:
+    1
+
+c_compiler:
+  - gcc                        # [linux]
+  - clang                      # [osx]
+  # part of a zip_keys: c_compiler, cxx_compiler, cuda_compiler_version
+  - vs2015                     # [win]
```
Comment (isuruf) [inline recipe/migrations/window_cuda.yaml:14]: We don't support vs2015 anymore

Comment (jaimergp) [reply recipe/migrations/window_cuda.yaml:14]: That will simplify the migration, but unfortunately it'll mean that CUDA 9.2 won't get support on Windows :/

Comment (isuruf) [reply recipe/migrations/window_cuda.yaml:14]: True, but @conda-forge/core made a decision to stop supporting it and going back is not possible without rebuilding all of the dependencies of a package needing CUDA 9.2.

Comment (jaimergp) [reply recipe/migrations/window_cuda.yaml:14]: The CUDA 9.2 headers [could be patched](https://github.com/LeelaChessZero/lczero.org/issues/12#issue-572284272), but that doesn't look like a good idea.

Comment (isuruf) [reply recipe/migrations/window_cuda.yaml]: Technically you can make it work by installing vc140 toolchain inside VS2017 and making sure `vs2015_win-64` use the VS2017 installation of vc140.

**Comment by jaimergp** [comment]
Is there anything else to be added here?

**Comment by CJ-Wright** [comment]
I think this looks ok, would you be able to baby sit things for a bit?

**Comment by jaimergp** [comment]
Babysitting as in helping other repos set up their Windows integrations?

**Comment by CJ-Wright** [comment]
More as in, watch the PRs the bot makes for this, if they seem incorrect (either in the build or the order) ping us and open a revert PR (or fix the migrator).

**Comment by jaimergp** [comment]
Sure, that I can do!

**Comment by CJ-Wright** [comment]
@isuruf is ok by you?

**Review by isuruf — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks for working on this Jaime! 😄

**Pull request merged**